client/internal/clientcommon.go

```diff
@@ -206,12 +206,7 @@ func (c *ClientCommon) PrepareFirstMessage(ctx context.Context) error {
 			msg.EffectiveConfig = cfg
 			msg.RemoteConfigStatus = c.ClientSyncedState.RemoteConfigStatus()
 			msg.PackageStatuses = c.ClientSyncedState.PackageStatuses()
-
-			if c.PackagesStateProvider != nil {
-				// We have a state provider, so package related capabilities can work.
-				msg.Capabilities |= protobufs.AgentCapabilities_AcceptsPackages
-				msg.Capabilities |= protobufs.AgentCapabilities_ReportsPackageStatuses
-			}
+			msg.Capabilities = c.Capabilities
```

Comment:
This ensures that the agent correctly reports the capabilities. I can extract this fix into a separate PR, but probably not worth it since it only became visible in this PR.